gcp/src/main/java/org/apache/iceberg/gcp/gcs/GCSOutputFile.java

```diff
@@ -67,6 +67,6 @@ public PositionOutputStream createOrOverwrite() {
 
   @Override
   public InputFile toInputFile() {
-    return new GCSInputFile(storage(), blobId(), gcpProperties(), metrics());
+    return new GCSInputFile(storage(), blobId(), null, gcpProperties(), metrics());
```

Comment:
Could we just use `GCSInputFile.fromLocation(...)` to avoid calling the constructor with null?

Reply:
I just didn't want to change very much here. We probably could.